src/cctag/Detection.cpp

```diff
@@ -102,7 +106,11 @@ static void constructFlowComponentFromSeed(
     }
     
     {
+#if TBB_VERSION_MAJOR > 2020
+      std::scoped_lock lock(G_SortMutex);
```

Comment:
another alternative is to use `boost::scoped_lock()` since Boost.Thread is already among the dependencies.
This would have the advantage of avoiding raising the standard to c++17.

Reply:
I think `std::lock_guard` would work just fine in this particular case.